### PR TITLE
Added INSTALL_OPTION docker build argument

### PR DESCRIPTION
**Pull request by sanal-vasudevan**
Added INSTALL_OPTION docker build argument to created docker image for any supported GoldenGate database type.

### CODE REVIEW

File: OracleGoldenGate/21c/README.md
Comment (Djelibeybi) [inline OracleGoldenGate/21c/README.md:48]: This should be Oracle Database 20c

File: OracleGoldenGate/21c/README.md
Comment (Djelibeybi) [inline OracleGoldenGate/21c/README.md:53]: Please replace the word "docker" with "container"

File: OracleGoldenGate/21c/README.md
Comment (Djelibeybi) [inline OracleGoldenGate/21c/README.md:62]: Please replace the word "docker" with "container"

**Review by Djelibeybi — CHANGES_REQUESTED** [review]
Please also update the copyright header for each of the modified files to 2022.

File: OracleGoldenGate/21c/README.md
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/README.md]: Done

File: OracleGoldenGate/21c/README.md
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/README.md]: Done

File: OracleGoldenGate/21c/README.md
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/README.md]: Done

**Comment by sbalousek** [comment]
The `INSTALLER` input should be used to determine the OGG variation being installed. Requiring the user to specify the additional `INSTALL_OPTION` input is awkward and error-prone.

The changes as-is are not acceptable.

**Comment by sbalousek** [comment]
> 



> @sbalousek as you are code owner, please submit a review that requests changes so that this PR is blocked until your review is cleared.

@Djelibeybi - Once we have a reasonable design, I definitely plan on reviewing the changes.

**Comment by sbalousek** [comment]
@sanal-vasudevan Examining the file name is one option. Another would be an examination of the contents in the installer. That was the method chosen for the 12c-19c solution. What makes no sense to me is the reason why the 'INSTALL_OPTION' value is even present in the response file. Is it really required?

**Comment by Djelibeybi** [comment]
> @Djelibeybi - Once we have a reasonable design, I definitely plan on reviewing the changes.

I need you to submit an empty review that requests changes as the code owner, because you are requesting changes, even if those haven't been defined yet. This is to prevent any accidental merges because GitHub will block merging until you approve the PR. 

There is no limit the amount of times you can review the same PR.

**Review by sbalousek — CHANGES_REQUESTED** [review]
This approach seems a little awkward, since the `INSTALLER` input already provides the data proposed by the `INSTALL_OPTION` input. Instead of requiring the user to provide the `INSTALL_OPTION` input, that would ideally be determined by either the file name in `INSTALLER` or there actual content of the file.

**Comment by sanal-vasudevan** [comment]
> @sanal-vasudevan Examining the file name is one option. Another would be an examination of the contents in the installer. That was the method chosen for the 12c-19c solution. What makes no sense to me is the reason why the 'INSTALL_OPTION' value is even present in the response file. Is it really required?

The 'INSTALL_OPTION' is a required property in the response file, if not specified, the installation will fail. 
I will try to derive the 'INSTALL_OPTION' automatically.

File: OracleGoldenGate/21c/README.md
Comment (sbalousek) [inline OracleGoldenGate/21c/README.md:27]: Please revert this whitespace-only change.

File: OracleGoldenGate/21c/README.md
Comment (sbalousek) [inline OracleGoldenGate/21c/README.md:42]: Place the example for BigData *after* the example for Oracle.

File: OracleGoldenGate/21c/bin/deployment-main.sh
Comment (sbalousek) [inline OracleGoldenGate/21c/bin/deployment-main.sh:73]: LIB_JVM_PATH must be `local`. A better name for this variable would be `JVM_LIBRARY_PATH`

File: OracleGoldenGate/21c/install-deployment.sh
Comment (sbalousek) [inline OracleGoldenGate/21c/install-deployment.sh:47]: This function violates all kinds of `shellcheck` rules. All shell scripts must pass `shellcheck` without warnings.

**Review by sbalousek — CHANGES_REQUESTED** [review]
So far, so good. The change to the shell scripts must pass `shellcheck` without warnings.

File: OracleGoldenGate/21c/bin/deployment-main.sh
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/bin/deployment-main.sh]: Done

Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/bin/deployment-main.sh]: Done

File: OracleGoldenGate/21c/README.md
Comment (sbalousek) [inline OracleGoldenGate/21c/README.md:38]: Revert this change and use the example for Oracle as a template for the BigData example.

Changes to markdown should pass 'markdown-lint'

File: OracleGoldenGate/21c/bin/deployment-main.sh
Comment (sbalousek) [inline OracleGoldenGate/21c/bin/deployment-main.sh:77]: Indentation in this file is four spaces.

File: OracleGoldenGate/21c/bin/deployment-main.sh
Comment (sbalousek) [inline OracleGoldenGate/21c/bin/deployment-main.sh:75]: Since `-e` is used, a missing `libjvm.so` will cause the script to terminate unexpectedly. Which means line 77 will never be executed.

Changes need to be tested!

File: OracleGoldenGate/21c/install-deployment.sh
Comment (sbalousek) [inline OracleGoldenGate/21c/install-deployment.sh:65]: Setting values for GLOBAL variables as a side effect of a function is a bad practice. Use echo and:

```
INSTALL_OPTION=$(ogg_install_option)
```

on line 80

File: OracleGoldenGate/21c/install-deployment.sh
Comment (sbalousek) [inline OracleGoldenGate/21c/install-deployment.sh:92]: Remove non-substantive change.

**Review by sbalousek — CHANGES_REQUESTED** [review]

File: OracleGoldenGate/21c/bin/deployment-main.sh
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/bin/deployment-main.sh:83]: Sorry,  that statement is not true in this case. 
Even if `-e` is used, the shell does not exit immediately if the command is part of the test in an if statement.

I tested these changes and works just fine.

File: OracleGoldenGate/21c/README.md
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/README.md]: Done.
In addition to this change, there were other markdown-lint corrections in this file suggested by this lint:  https://github.com/markdownlint/markdownlint

File: OracleGoldenGate/21c/install-deployment.sh
Comment (sbalousek) [inline OracleGoldenGate/21c/install-deployment.sh:66]: This function is not very succinct. A more efficient implementation is:

```.sh
function ogg_install_option() {
    awk '/INSTALL_TYPE/ {
            sub(/^.*INSTALL_TYPE="/, "")   
            sub(/".*/, "")
            print
            exit 0
        }' $(find /tmp/installer -name fastcopy.xml -print -quit)
}
```

**Review by sbalousek — CHANGES_REQUESTED** [review]

File: OracleGoldenGate/21c/README.md
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/README.md:18]: Okay, I will revert.

Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/README.md:18]: I tried to make the corrections as per this lint: https://github.com/markdownlint/markdownlint

It will be helpful if you can please point to the correct `markdown-lint` that should be used for any future changes.

File: OracleGoldenGate/21c/README.md
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/README.md:25]: Okay

File: OracleGoldenGate/21c/install-deployment.sh
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/install-deployment.sh:66]: Thanks! Steve, this looks really good.

There is one issue with this function, in case the `find` command does not return any output, `awk` will just wait there indefinitely expecting an input file.  I will try to address that.

Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/install-deployment.sh:66]: Btw, another implementation similar to yours using `perl` which IMO is more succinct. 


```bash
function ogg_install_option() {
    local fastcopy_file
    fastcopy_file=$(find /tmp/installer -name fastcopy.xml -print -quit)
    [[ -z "${fastcopy_file}" ]] && abort "The file 'fastcopy.xml' could not be located."
    perl -ne '/^.* INSTALL_TYPE="(.*)" .*$/ && print $1' "${fastcopy_file}"
}
```

File: OracleGoldenGate/21c/install-deployment.sh
Comment (sbalousek) [inline OracleGoldenGate/21c/install-deployment.sh:55]: Perl is not used in this project.

File: OracleGoldenGate/21c/install-deployment.sh
Comment (sbalousek) [reply OracleGoldenGate/21c/install-deployment.sh:56]: We never, ever use Perl.

File: OracleGoldenGate/21c/install-deployment.sh
Comment (sbalousek) [inline OracleGoldenGate/21c/install-deployment.sh:69]: Use:

```sh
INSTALL_OPTION=${INSTALL_OPTION:-$(ogg_install_option)}
```

Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/install-deployment.sh]: Done.

File: OracleGoldenGate/21c/install-deployment.sh
Comment (sanal-vasudevan) [reply OracleGoldenGate/21c/install-deployment.sh]: Thank you for the information. I have modified this to use `awk`

**Review by sbalousek — APPROVED** [review]
These changes look great. Thanks, Sanal, for the contribution.

**Review by Djelibeybi — APPROVED** [review]
@sbalousek is happy, which makes me happy.

**Comment by sanal-vasudevan** [comment]
> These changes look great. Thanks, Sanal, for the contribution.

Happy that I could contribute, and thank you @sbalousek

**Pull request merged**
